### PR TITLE
Add initial Editors and Tools page

### LINKED ISSUE #34: Tools- editors, plugins etc
A page dedicated just to Code Editors and the most popular plugins for them.

### PR DESCRIPTION
**Pull request by dobromir-hristov**
This PR adds basic info about IDEs and tools for Vue Development. 

The idea is to expose the most popular solutions and guide people towards what to choose. We are getting lots of questions on whats "best", we should not answer that, but at least guide them towards an answer.

In the Tools section I will explain very very briefly about ESLint and Prittier, and where the confusion comes from whem using them together and how to resolve it + some nice articles on the matter. Do you have any other tools we could add here?

Please take a look and tell me if the direction is right for you.

Closes #34

### CODE REVIEW

File: docs/guide/ecosystem/editors-and-tools.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+As a fully featured IDE, WebStorm provides greater refactoring capabilities, improved code completion, hinting and highlighting, but at a cost of speed. It is heavy and can get choppy if you open too many files or projects at the same time.
+
+### CodeSandbox <badge text="Rising Star"/>
+
+CodeSandbox is the only online, cloud based code editor, that supports Vue's SFC format, among many other frameworks and popular client apps and server setups. Vue projects are built with Vue CLI, giving you a well known folder structure to extend upon. 
+
+Based on VSCode, CodeSandbox has a similar look and feel. When you create a Vue project, you will benefit from code completion, hinting, syntax highlighting and more. Apart from Vue CLI, they have templates for [Vuepress](../ecosystem/documentation.md#vuepress) and [Nuxt](../ecosystem/nuxt.md), just to name a few.
+
+CodeSandbox has a very generous freemium policy with the first 50 boxes (projects) absolutely free.
```
Comment (gustojs) [inline docs/guide/ecosystem/editors-and-tools.md]: Not really the first 50 - just 50 at the same time. Once you reach the limit, you can delete old unused boxes and create new ones.

File: docs/guide/ecosystem/editors-and-tools.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+- [Official Website](https://codesandbox.net)
+
+</useful-links-section>
+</useful-links>
+
+### Atom
+
+### SublimeText
+
+Another very popular editor, coming close to VSCode. It is well known for its speed and lightness, as well as its free to use model. Sublime has a very big library of free plugins, extensions and themes, enabling far more features, than was intended in the beginning.
```
Comment (gustojs) [inline docs/guide/ecosystem/editors-and-tools.md]: Sublime Text is not free to use - you can download it and play with it but for continuous use, you're required to buy a licence.

File: docs/guide/ecosystem/editors-and-tools.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+To enable Vue support you will need to install the official, free Vue.js plugin from the plugins menu. It will give you `.vue` file support with highlighting, `template` and `script` tag code completion, linting, component hinting, prop completion, type hinting, refactoring support and much more.
+
+As a fully featured IDE, WebStorm provides greater refactoring capabilities, improved code completion, hinting and highlighting, but at a cost of speed. It is heavy and can get choppy if you open too many files or projects at the same time.
+
+### CodeSandbox <badge text="Rising Star"/>
+
+CodeSandbox is the only online, cloud based code editor, that supports Vue's SFC format, among many other frameworks and popular client apps and server setups. Vue projects are built with Vue CLI, giving you a well known folder structure to extend upon. 
+
+Based on VSCode, CodeSandbox has a similar look and feel. When you create a Vue project, you will benefit from code completion, hinting, syntax highlighting and more. Apart from Vue CLI, they have templates for [Vuepress](../ecosystem/documentation.md#vuepress) and [Nuxt](../ecosystem/nuxt.md), just to name a few.
```
Comment (gustojs) [inline docs/guide/ecosystem/editors-and-tools.md]: I'd add somewhere that we can integrate Codesandbox with Github (or Git in overall?). Imho that's the main feature that makes it fit this category at all.

File: docs/guide/ecosystem/editors-and-tools.md [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+
+To enable Vue support you will need to install the official, free Vue.js plugin from the plugins menu. It will give you `.vue` file support with highlighting, `template` and `script` tag code completion, linting, component hinting, prop completion, type hinting, refactoring support and much more.
+
+As a fully featured IDE, WebStorm provides greater refactoring capabilities, improved code completion, hinting and highlighting, but at a cost of speed. It is heavy and can get choppy if you open too many files or projects at the same time.
+
+### CodeSandbox <badge text="Rising Star"/>
+
+CodeSandbox is the only online, cloud based code editor, that supports Vue's SFC format, among many other frameworks and popular client apps and server setups. Vue projects are built with Vue CLI, giving you a well known folder structure to extend upon.
```
[the remark this answers is not in the document]
Comment (dobromir-hristov) [reply docs/guide/ecosystem/editors-and-tools.md]: No vue project in there.

File: docs/guide/ecosystem/editors-and-tools.md [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+<useful-links>
+<useful-links-section title="Useful Links">
+
+- [Official Website](https://codesandbox.net)
+
+</useful-links-section>
+</useful-links>
+
+### Atom
+
```
Comment (shentao) [inline docs/guide/ecosystem/editors-and-tools.md]: An amazing free, open-source code editor created by GitHub with plenty of extensions and superb user experience. With the available plugins, it supports a wide range of languages and features, including ones dedicated to Vue: smart code completion/suggestions, snippets, syntax highlighting, linting and formatting with both eslint and prettier, that works in template, script and style parts of the Vue SFC.

On top of that, Atom has great integration with GitHub when it comes to managing commits, PRs, reviewing changes and can be further extended with additional plugins. Some say its the most fashionable and good looking editor out there and the code completion is at the same time very smart (thanks to Tern.js), but also completely unintrusive. You’re the developer, you got this!

Some say, that it’s a bit slow, but that’s just a way of hinting that you should refactor your huge components.

<useful-links>
<useful-links-section title="Useful Links">

- [ternjs](https://github.com/tststs/atom-ternjs)
- [language-vue](https://github.com/hedefalk/atom-vue)
- [ide-vue](https://github.com/rwatts3/atom-ide-vue)
- [vue-fmt](https://github.com/voodeng/vue-fmt)

</useful-links-section>
</useful-links>

Comment (gustojs) [reply docs/guide/ecosystem/editors-and-tools.md]: Imho this section feels pretty opinionated. There are parts of the guide where we focus on opinions (like tips) but the ecosystem one should give an objective overview feel.

Comment (dobromir-hristov) [reply docs/guide/ecosystem/editors-and-tools.md]: It's written by the only one using it, what do you expect? 😆

Comment (shentao) [reply docs/guide/ecosystem/editors-and-tools.md]: Oh. So why don’t we sort the elements in the guide alphabetically? Why does VS Code have the "hottest" icon? Why isn’t Atom listed in the "best" code editors? 
My point is, the whole thing is already opinionated. 😉 
The difference here is that I decided to actually add some personality into the description in hopes it makes for a more engaging content. If you don’t agree or like it, feel free to redact it – I won’t mind!

**Pull request merged**